packages/ui/components/Form/Inputs/Dropdown.tsx

```diff
@@ -0,0 +1,125 @@
+import * as RadixSelect from '@radix-ui/react-select'
+import { CheckIcon, ChevronDownIcon, ChevronUpIcon } from '../../icons'
+import classnames from 'classnames'
+
+type DropdownRegularOption = {
+  type?: 'regular'
+  value: string
+  label: string
+}
+
+type DropdownRegularOptionProps = {
+  option: DropdownRegularOption
+}
+
+const DropdownRegularOption = ({ option: { value, label } }: DropdownRegularOptionProps) => {
+  return (
+    <RadixSelect.Item
+      className="flex items-center relative text-gray-200 text-sm leading-7 pl-6 pr-7 hover:bg-gray-700 focus:bg-gray-700 my-2 rounded outline-none"
+      value={value}
+    >
+      <RadixSelect.ItemText>{label}</RadixSelect.ItemText>
+      <RadixSelect.ItemIndicator className="absolute left-0 inline-flex items-center justify-center">
+        <CheckIcon className="w-4 h-4 pl-1" />
+      </RadixSelect.ItemIndicator>
+    </RadixSelect.Item>
+  )
+}
+
+type DropdownGroupOption = {
+  type: 'group'
+  label: string
+  options: DropdownRegularOption[]
+}
+
+type DropdownGroupOptionProps = {
+  option: DropdownGroupOption
+}
+
+const DropdownGroupOption = ({ option: { label, options } }: DropdownGroupOptionProps) => {
+  return (
+    <RadixSelect.Group>
+      <RadixSelect.Label className="text-xs text-gray-500 px-4 leading-6">{label}</RadixSelect.Label>
+      {options.map((option) => (
+        <DropdownRegularOption option={option} key={option.value} />
+      ))}
+    </RadixSelect.Group>
+  )
+}
+type DropdownSeparatorOption = {
+  type: 'separator'
+}
+
+const DropdownSeparatorOption = () => {
+  return <RadixSelect.Separator className="w-full h-px bg-gray-700 my-2" />
+}
+
+export type DropdownOption = DropdownGroupOption | DropdownRegularOption | DropdownSeparatorOption
+
+type DropdownOptionProps = {
+  option: DropdownOption
+}
+
+const DropdownOption = ({ option }: DropdownOptionProps) => {
+  switch (option.type) {
+  case 'separator':
+    return <DropdownSeparatorOption />
+  case 'group':
+    return <DropdownGroupOption option={option} />
+  default:
+    return <DropdownRegularOption option={option} />
+  }
+}
+
+export type DropdownProps = {
+  options: DropdownOption[]
+  value?: string
+  onChange?: (selectedOption: string) => void
+  placeholder?: string
+  isDisabled?: boolean
+  name?: string
+  className?: string
+}
+
+export function Dropdown({
+  options,
+  value,
+  onChange,
+  placeholder,
+  name,
+  isDisabled,
+  className,
+}: DropdownProps) {
```

Comment:
```suggestion
}: DropdownProps) {
  placeholder = (!options || options.length === 0) ? "No protocols available..." : "Select an option...";
```

- We can provide a **placeholder value here.** This way, **we won't need to define it every time in the stories.**

Reply:
I like that we have a placeholder, although it has to come as a prop and not as a variable inside the component. Maybe this is what you meant?

```suggestion
  placeholder = 'Select an option...',
}: DropdownProps) {
```

Since this is a generic dropdown component that will be used in multiple contexts, I'd avoid using references to protocols and specific things. This information should come from the user of the component, not from the component itself.

Reply:
Got it. It is a generic dropdown component. 
> Maybe this is what you meant?
`  placeholder = 'Select an option...',`

Yes, perfect 👌